pkg/operator/k8sutil/node.go

```diff
@@ -321,6 +322,15 @@ func RookNodesMatchingKubernetesNodes(rookStorage cephv1.StorageScopeSpec, kuber
 
 // GenerateNodeAffinity will return v1.NodeAffinity or error
 func GenerateNodeAffinity(nodeAffinity string) (*v1.NodeAffinity, error) {
+	if json.Valid([]byte(nodeAffinity)) {
```

Comment:
Is only JSON supported? Yaml is just more common in our examples.

Reply:
Added YAML support.